php-checks/src/main/java/org/sonar/php/checks/CountInsteadOfEmptyCheck.java

```diff
@@ -89,6 +91,90 @@ private boolean isCountFunction(FunctionCallTree tree) {
     return FUNCTION_PREDICATE.test(TreeValues.of(tree, context().symbolTable()));
   }
 
+  private static final Tree.Kind[] CONDITIONAL_KINDS = {
+    Tree.Kind.CONDITIONAL_AND,
+    Tree.Kind.CONDITIONAL_OR,
+    Tree.Kind.ALTERNATIVE_CONDITIONAL_AND,
+    Tree.Kind.ALTERNATIVE_CONDITIONAL_OR,
+    Tree.Kind.ALTERNATIVE_CONDITIONAL_XOR,
+  };
+
+  private static final Tree.Kind[] CONDITION_STATEMENTS = {
+    Tree.Kind.PARENTHESISED_EXPRESSION,
+    Tree.Kind.CONDITIONAL_EXPRESSION
+  };
+
+  /**
+   * Checks if "empty" is called for the given variable in the current condition.
+   * For example, in case of "empty($foo) && count($foo) > 0" we want to know if "empty" is called in addition to "count".
+   * @param countCallTree represents the count call in the current condition.
+   * @param countArgument the argument passed to the call of count.
+   * @return true if there exists an "empty" call for the countArgument in the scope of the current condition.
+   */
+  private boolean isEmptyUsedInCondition(FunctionCallTree countCallTree, @Nullable ExpressionTree countArgument) {
+    if (!(countArgument instanceof VariableIdentifierTreeImpl countArgumentVariable)) {
+      // "count" is called without an argument
+      return false;
+    }
+
+    Tree currentCondition = countCallTree.getParent();
+
+    do {
+      currentCondition = currentCondition.getParent();
```

Comment:
I think we are unintentionally skipping the first parent here:

```suggestion
    Tree currentCondition = countCallTree;

    do {
      currentCondition = currentCondition.getParent();
```